src/buckets/components/context/csvUploaderProvider.tsx

```diff
@@ -138,9 +164,22 @@ export const CsvUploaderProvider: FC<Props> = React.memo(({children}) => {
             }
             time = table.getColumn('_time')?.[i] ?? Date.now()
             value = table.getColumn('_value')?.[i] ?? field ?? ''
+            // Adds quotes to values if _value is of string type
+            // https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/#quotes
+            value =
+              table.getColumnType('_value') === 'string' && value
+                ? '"' + value + '"'
+                : value
             tags = columns
               .filter(col => !!table.getColumn(col)[i])
-              .map(col => `${col}=${table.getColumn(col)[i]}`)
+              .map(
+                col =>
+                  `${col}=${
+                    table.getColumnType(col) === 'string'
+                      ? parseTagStringValue(table.getColumn(col)[i] as string)
```

Comment:
This is solid, and under normal circumstances I wouldn't bat an eyelash at having an abstracted function that serves a specific purpose (in this case, formatting a value to align with our write endpoint expectations). However, since we are looping over this function N times, we are creating N instances of this function. In this case, readability provides a hit to performance. Would love it if we could simply inline the regex with a comment

Reply:
nice find!